rules/aws_s3_bucket_name.go

```diff
@@ -51,6 +64,68 @@ func (r *AwsS3BucketNameRule) Link() string {
 	return project.ReferenceLink(r.Name())
 }
 
+// Return regex bucket naming rules
+func rules() ([]map[string]string, error) {
+	// Get bucket naming rule based on length from AWS SDK API models
+	r, err := http.Get(
+		"https://raw.githubusercontent.com/aws/aws-sdk-go/main/models/apis/cloudformation/2010-05-15/api-2.json",
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Body.Close()
+
+	bucket := Bucket{}
+	err = json.NewDecoder(r.Body).Decode(&bucket)
+	if err != nil {
+		return nil, err
+	}
+
+	// https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+	return []map[string]string{
+		{
+			"regx": fmt.Sprintf(
+				"^(.{0,%d})?(.{%d,})?$",
+				bucket.Shapes.S3Bucket.Min-1,
+				bucket.Shapes.S3Bucket.Max+1,
+			),
+			"desc": fmt.Sprintf(
+				"Bucket names must be between %d (min) and %d (max) characters long.",
+				bucket.Shapes.S3Bucket.Min,
+				bucket.Shapes.S3Bucket.Max,
+			),
+		},
+		{
+			"regx": "[^a-z0-9\\.\\-]",
+			"desc":  "Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).",
+		},
+		{
+			"regx": "^[^a-z0-9]",
+			"desc":  "Bucket names must begin with a lowercase letter or number.",
+		},
+		{
+			"regx": "[^a-z0-9]$",
+			"desc":  "Bucket names must end with a lowercase letter or number.",
+		},
+		{
+			"regx": "\\.\\.",
+			"desc":  "Bucket names must not contain two adjacent periods.",
+		},
+		{
+			"regx": "^\\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\.|$)){4}\\b$",
```

Comment:
Any citation for this? A simple `^(\d{1,3}\.){3}\d{1,3}$` needs none but there is much more logic here.

Reply:
192.168.0.1 -> Not OK
393.388.2.321 -> OK

Reply:
Sure I get it, but it's still a complicated regex. An alternative:

https://pkg.go.dev/net#ParseIP

If it returns non nil and calling `To4` on the result is also non-nil, you know it's an IPv4 address. That code is easier to read than a regular expression.

Reply:
It'd have to be a separate check other than regex, then.

Reply:
```go
nameAsIP := net.ParseIP(name)
if nameAsIP != nil && net.IP.To4(nameAsIP) != nil {
  runner.EmitIssue(
    r,
    fmt.Sprintf(`Bucket names must not be formatted as an IP address. (name: %q)`, name),
    attribute.Expr.Range(),
  )
}
```

Reply:
Yep, that looks good to me!